main.js

```diff
@@ -1 +1,115 @@
-console.log("Tic Tac Toe")
\ No newline at end of file
+// alternatives for var?
+var grid, playerActive, gameGoing, counts;
+
+function makeArray(w, h, val) {
+  result = [];
+  for (let i = 0; i <= h; i++) {
+    let temp = [];
+    result.push(temp);
+    for (let j = 0; j <= w; j++) {
+      temp.push(val);
+    }
+  }
+  return result;
+}
+
+//Restart everything
+const init = function() {
+  counts = 0;
+  grid = makeArray(2, 2, -1);
+  playerActive = 0;
+  gameGoing = true;
+  document.querySelector(".p" + 0 + "bar").style.display = "block";
+  document.querySelector(".p" + 1 + "bar").style.display = "none";
+  document.querySelector(".w" + 1).style.display = "none";
+  document.querySelector(".w" + 0).style.display = "none";
+
+  let imgs = document.querySelector(".images");
+  while (imgs) {
+    imgs.parentNode.removeChild(imgs);
+    imgs = document.querySelector(".images");
+  }
+  let draw = document.querySelector(".draw");
+  if (draw) {
+    draw.parentNode.removeChild(draw);
+  }
+};
+
+//Updating game score and check for winning condition
+const algorithm = function(j, i) {
+  grid[j][i] = playerActive;
+  //column-wise
+  if (
+    grid.slice(j - 1)[0].slice(i)[0] === playerActive &&
+    grid.slice((j + 1) % grid.length)[0].slice(i)[0] === playerActive
+  ) {
+    return true;
+  }  
+  //row-wise
+  if (
+    grid.slice(j)[0].slice(i - 1)[0] === playerActive &&
+    grid.slice(j)[0].slice((i + 1) % grid.length)[0] === playerActive
+  ) {
+    return true;
+  }
+  //diagnal wise top-down
+  if (grid[0][0] == playerActive && grid[1][1] == playerActive && grid[2][2] == playerActive) {
+    return true;
+  }
+  //diagnal wise bottom-up
+  if (grid[0][2] == playerActive && grid[1][1] == playerActive && grid[2][0] == playerActive) {
+    return true;
+  }
+  return false;
+  
+};
+
+// Listen and animate the playing process
+const animate = function(index) {
+  document.querySelector("#" + index).addEventListener("click", function() {
+    if (gameGoing) {
+      counts += 1;
+      const i = index.slice(-1)[0];
+      const j = index.slice(-2)[0];
```

Comment:
This looks ugly. I understand what you are trying to achieve here but I think you can pass a second parameter to your `animate` function to accept grid item number